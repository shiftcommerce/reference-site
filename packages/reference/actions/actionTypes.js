// Action types
export const SET_PRODUCT = 'SET_PRODUCT'
export const UPDATE_CART_LINE_ITEMS = 'UPDATE_CART_LINE_ITEMS'
export const STORE_CART = 'STORE_CART'
export const SET_SEARCH_STATE = 'SET_SEARCH_STATE'
export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY'
export const SET_CATEGORY = 'SET_CATEGORY'
export const SET_CATEGORIES = 'SET_CATEGORIES'
export const SET_CHECKOUT_INPUT_VALUE = 'SET_CHECKOUT_INPUT_VALUE'
export const SET_CHECKOUT_INPUT_VALIDATION_MESSAGE = 'SET_CHECKOUT_INPUT_VALIDATION_MESSAGE'
export const STORE_CHECKOUT = 'STORE_CHECKOUT'
export const SET_CHECKOUT_INPUT_SHOWN = 'SET_CHECKOUT_INPUT_SHOWN'
export const SET_SHIPPING_METHOD = 'SET_SHIPPING_METHOD'
export const TOGGLE_CHECKOUT_COMPONENT_COLLAPSED = 'TOGGLE_CHECKOUT_COMPONENT_COLLAPSED'
export const CREATE_ORDER = 'CREATE_ORDER'
export const SET_SHIPPING_AS_BILLING_KEY = 'SET_SHIPPING_AS_BILLING_KEY'
export const INITIATE_CHECKOUT = 'INITIATE_CHECKOUT'
export const INITIATE_CART = 'INITIATE_CART'
export const CHANGE_PAYMENT_METHOD = 'CHANGE_PAYMENT_METHOD'
export const REQUEST_CARD_TOKEN = 'REQUEST_CARD_TOKEN'
export const SET_CARD_TOKEN = 'SET_CARD_TOKEN'
export const SET_PAYMENT_ERROR = 'SET_PAYMENT_ERROR'
