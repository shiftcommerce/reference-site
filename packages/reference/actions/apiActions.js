// Libs
import ApiClient from '../lib/ApiClient'
import JsonApiParser from '../lib/JsonApiParser'

export const readEndpoint = (request) => {
  return (dispatch, getState) => {
    dispatch(setLoadingTo(true, request.successActionType))

    return new ApiClient().read(request.endpoint, request.query)
      .then((response) => {
        if (response.status === 200 || response.status === 304) {
          const parsedPayload = new JsonApiParser().parse(response.data)
          dispatch({ type: request.successActionType, payload: Object.assign({}, parsedPayload, {loading: false}) })
        } else {
          dispatch(setLoadingTo(false, request.successActionType))
          dispatch(setErroredTo(true, request.successActionType))
        }
      })
      .catch(() => {
        dispatch(setLoadingTo(false, request.successActionType))
        dispatch(setErroredTo(true, request.successActionType))
      })
  }
}

export const postEndpoint = (request) => {
  return (dispatch, getState) => {
    dispatch(setLoadingTo(true, request.successActionType))
    return new ApiClient().post(request.endpoint, request.body)
      .then((response) => {
        if (response.status === 200) {
          const parsedData = JSON.parse(response.data.data)
          const parsedPayload = new JsonApiParser().parse(parsedData.data)
          dispatch({ type: request.successActionType, payload: Object.assign({}, parsedPayload, {loading: false}) })
        } else {
          dispatch(setLoadingTo(false, request.successActionType))
          dispatch(setErroredTo(true, request.successActionType))
        }
      })
      .catch(() => {
        dispatch(setLoadingTo(false, request.successActionType))
        dispatch(setErroredTo(true, request.successActionType))
      })
  }
}

function setLoadingTo (boolean, actionType) {
  return {
    type: actionType,
    payload: {
      loading: boolean
    }
  }
}

function setErroredTo (boolean, actionType) {
  return {
    type: actionType,
    payload: {
      error: boolean
    }
  }
}
