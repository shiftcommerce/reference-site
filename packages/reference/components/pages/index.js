import CategorySidebar from './categorySidebar'

const PageTemplates = {
  category_sidebar_template: CategorySidebar
}

export default PageTemplates
