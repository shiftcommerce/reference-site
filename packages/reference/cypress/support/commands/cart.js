/**
 * Adds variant to cart
 */
Cypress.Commands.add('addVariantToCart', () => {
  // Setup server
  cy.server()

  // Stub add to cart request
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-13.json'
  }).as('addToCart')

  // Navigate to PDP
  cy.visit('/products/clock_computer')
  // Select the instock variant
  cy.contains(/Clock Computer 13'' - £799.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCart')
})

/**
 * Adds variants to cart
 */
Cypress.Commands.add('addVariantsToCart', () => {
  // Setup server
  cy.server()

  // Stub first add to cart requests
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-13.json'
  }).as('addToCartClockComputer13')

  // Navigate to PDP
  cy.visit('/products/clock_computer')
  // Select the first variant
  cy.contains(/Clock Computer 13'' - £799.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCartClockComputer13')
  // Click the cross to close minibag
  cy.get('.c-minibag__dropdown-cross').click()

  // Stub second add to cart requests
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-15.json'
  }).as('addToCartClockComputer15')

  // Select the second variant
  cy.contains(/Clock Computer 15'' - £999.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCartClockComputer15')
})

/**
 * Adds promotion code to cart
 * @param {string} promoCode - the promotion code
 */
Cypress.Commands.add('addPromotionCodeToCart', ({ promoCode }) => {
  // Check the promotion form title
  cy.get('.c-coupon-form__title').contains(/Promotion Code/i)
  // Enter a promotion code
  cy.get('.c-coupon-form__input').type(promoCode)
  // Click apply button
  cy.get('.c-coupon-form__button').click()
})
