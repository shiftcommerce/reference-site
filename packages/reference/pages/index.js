import { Component } from 'react'

// components
import Layout from '../components/Layout'

class Home extends Component {
  render () {
    return <Layout>
      <div>Welcome To Shift Commerce Front End React</div>
    </Layout>
  }
}

export default Home
