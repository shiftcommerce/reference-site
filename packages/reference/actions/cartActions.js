// actionTypes
import * as actionTypes from './actionTypes'

// cartHandler
import * as cartHandler from './handlers/cartHandler'

// initial states
import { cartInitialState } from './../reducers/setCart'
import { checkoutInitialState } from './../reducers/setCheckout'

// localForage
import LocalForage from '../lib/localforage'
const localForage = new LocalForage()

// Return to reducer with the updated cart details
export function updateCart (cart) {
  return {
    type: actionTypes.UPDATE_CART_LINE_ITEMS,
    cart: Object.assign({}, cart, {loading: false})
  }
}

// Store the cart details in to the local redux store
export function storeCart (cart) {
  return {
    type: actionTypes.STORE_CART,
    cart: Object.assign({}, cart, {loading: false})
  }
}

export function initiateCart () {
  return {
    type: actionTypes.INITIATE_CART
  }
}

export function initiateCheckout () {
  return {
    type: actionTypes.INITIATE_CHECKOUT
  }
}

// This function will be called by minibag on every page load
// It will initialize cart lineitems
export function readCart () {
  return (dispatch, getState) => {
    let cart = getState().cart

    // check if local redux has line items
    if (cart.totalQuantity === 0) {
      // check if indexedDB, if local redux store dont have line item
      // This is to ensure, we dont miss line items on page refresh
      return localForage.getValue('cart').then((cart) => {
        // if indexedDB also dont have lineitems,
        // initalize cart
        if (cart === null) {
          dispatch(initializeCart())
        }
        // update local redux store
        dispatch(storeCart(cart))

        return cart
      })
    // if local redux store has line items,
    // dispatch them
    } else {
      dispatch(storeCart(cart))
      return Promise.resolve(cart)
    }
  }
}

// This function will be called, when user add items to bag
// in product detail page. If a line item already exists, it
// will add the new quantity to the existing one
export function addToCart (lineItem) {
  return (dispatch, getState) => {
    let cart = getState().cart
    addOrUpdateLineItems(cart, lineItem, dispatch)
  }
}

export function updateQuantity (lineItem) {
  return (dispatch, getState) => {
    let cart = getState().cart
    cart = cartHandler.updateCart(cart, lineItem)
    dispatchUpdate(cart, dispatch)
  }
}

export function initializeCart () {
  return (dispatch) => {
    dispatch(initiateCart())
    dispatch(initiateCheckout())
    localForage.setValue('cart', cartInitialState)
    localForage.setValue('checkout', checkoutInitialState)
  }
}

// Private functions

function addOrUpdateLineItems (cart, lineItem, dispatch) {
  cart = cartHandler.addToCart(cart, lineItem)
  dispatchUpdate(cart, dispatch)
}

function dispatchUpdate (cart, dispatch) {
  // update local redux store
  dispatch(updateCart(cart))
  // update indexedDB
  localForage.setValue('cart', cart)
}
