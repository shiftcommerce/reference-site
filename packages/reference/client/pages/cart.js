// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Head from 'next/head'

// Lib
import { algoliaReduxWrapper } from '../lib/algolia-redux-wrapper'
import { suffixWithStoreName } from '../lib/suffix-with-store-name'

// Actions
import { updateQuantity } from '../actions/cart-actions'

// Components
import CartTable from '../components/cart/cart-table'
import Loading from '../components/loading'

export class CartPage extends Component {
  constructor (props) {
    super(props)

    this.updateQuantity = this.updateQuantity.bind(this)
    this.deleteItem = this.deleteItem.bind(this)
  }

  updateQuantity (e) {
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: parseInt(e.target.value, 10)
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  deleteItem (e) {
    e.preventDefault()
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: 0
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  render () {
    const { cart, cart: { loading, error } } = this.props

    if (loading) {
      return (
        <Loading />
      )
    } else if (error) {
      return (
        <p>{ error }</p>
      )
    } else {
      return (
        <>
          <Head>
            <title>{ suffixWithStoreName('Your Shopping Cart') }</title>
          </Head>
          <CartTable cart={cart} updateQuantity={this.updateQuantity} deleteItem={this.deleteItem} />
        </>
      )
    }
  }
}

function mapStateToProps (state) {
  const { cart } = state
  return { cart }
}

export default algoliaReduxWrapper(connect(mapStateToProps)(CartPage), CartPage)
