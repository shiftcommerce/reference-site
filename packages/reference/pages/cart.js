// Libraries
import { Component } from 'react'
import withRedux from 'next-redux-wrapper'

// Utils
import { configureStore } from '../utils/configureStore'

// Actions
import { updateQuantity } from '../actions/cartActions'
import { readMenu } from '../actions/menuActions'

// Components
import Layout from '../components/Layout'
import CartTable from '../components/cart/CartTable.js'

export class CartPage extends Component {
  static async getInitialProps ({ store }) {
    await store.dispatch(readMenu(store))
  }

  constructor (props) {
    super(props)

    this.updateQuantity = this.updateQuantity.bind(this)
  }

  updateQuantity (e) {
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: parseInt(e.target.value, 10)
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  render () {
    const cart = this.props.cart

    if (cart.loading) {
      return (
        <Layout>
          <p>loading...</p>
        </Layout>
      )
    } else if (cart.error) {
      return (
        <Layout>
          <p>{cart.error}</p>
        </Layout>
      )
    } else {
      return (
        <Layout>
          <CartTable cart={cart} updateQuantity={this.updateQuantity} />
        </Layout>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    cart: state.cart || {}
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: dispatch
  }
}

export default withRedux(configureStore, mapStateToProps, mapDispatchToProps)(CartPage)
