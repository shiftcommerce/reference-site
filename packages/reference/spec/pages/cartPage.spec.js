// Libraries
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'

// Pages
import { CartPage } from '../../pages/cart'

// Actions
import { updateQuantity } from '../../actions/cartActions'

// Fixtures
import cart from '../fixtures/cart.fixture'
import menu from '../fixtures/menu.fixture'

test('dispatch updateQuantity action on changing line item quantity', () => {
  // Arrange
  const initialState = {
    menu: {
      loading: false,
      error: false,
      data: menu
    }
  }
  const expectedFunction = updateQuantity().toString()
  const updateQuantitySpy = jest.spyOn(CartPage.prototype, 'updateQuantity')
  const dispatch = jest.fn().mockImplementation((updateSpy) => 'first call')

  // Act
  const wrapper = mount(
    <Provider store={createMockStore(initialState)}>
      <CartPage cart={cart} dispatch={dispatch} />
    </Provider>
  )

  // Assert
  expect(wrapper).toMatchSnapshot()

  // Verify if cart line items are available
  expect(wrapper).toIncludeText('2 items in your shopping basket')

  // To clear the logs of dispatch being called on component mount
  dispatch.mockClear()

  // Trigger quantity change
  wrapper.find('select').simulate('change', {target: { value: 3, dataset: { variant: '123' } }})

  // Verify if updateQuantity method is getting triggered on quantity update
  expect(updateQuantitySpy).toHaveBeenCalled()

  expect(dispatch).toHaveBeenCalled()

  // Verify it dispatch method called a function
  expect(dispatch.mock.calls[0][0]).toEqual(expect.any(Function))

  // Verify if the dispatch function has dispatched updateQuantity action.
  expect(dispatch.mock.calls[0][0].toString()).toMatch(expectedFunction)
})
