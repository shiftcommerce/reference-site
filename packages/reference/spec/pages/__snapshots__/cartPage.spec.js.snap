// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatch updateQuantity action on changing line item quantity 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartPage
    cart={
      Object {
        "lineItems": Array [
          Object {
            "discount": 0,
            "imageUrl": "",
            "price": 10,
            "productID": 1,
            "productSku": "1231",
            "quantity": 2,
            "size": "size - 8",
            "sku": "123",
            "title": "test",
          },
        ],
      }
    }
    dispatch={[Function]}
  >
    <Layout>
      <div>
        <CustomHead>
          <SideEffect(Head)>
            <Head />
          </SideEffect(Head)>
        </CustomHead>
        <div
          className="o-header"
        >
          <span
            className="o-header__logo"
          >
            <Link
              as="/"
              href="/home/index"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <Image
                  height={59}
                  src="/static/logo.svg"
                  width={52}
                >
                  <div
                    className={undefined}
                    style={
                      Object {
                        "height": 59,
                        "width": 52,
                      }
                    }
                  >
                    <img
                      className="c-image"
                      height={59}
                      src="/static/logo.svg"
                      width={52}
                    />
                  </div>
                </Image>
              </a>
            </Link>
          </span>
          <span
            className="o-header__search"
          >
            <SearchBar
              onChange={undefined}
              queryObject={undefined}
            >
              <form
                action="/search"
                method="get"
                onSubmit={[Function]}
                role="search"
              >
                <div
                  className="c-searchbar"
                >
                  <input
                    aria-label="search text"
                    className="c-searchbar--input"
                    name="query"
                    onChange={[Function]}
                    placeholder="Enter Keyword Or Product Code"
                    type="search"
                    value=""
                  />
                  <Button
                    label="Search"
                    size="lrg"
                    status="dark"
                  >
                    <button
                      className="o-button o-button--dark o-button--lrg"
                      role="button"
                      tabIndex="0"
                    >
                      Search
                    </button>
                  </Button>
                </div>
              </form>
            </SearchBar>
          </span>
          <span
            className="o-header__minibag"
          >
            <Connect(MiniBag)>
              <MiniBag
                cart={Object {}}
                dispatch={[Function]}
              >
                <div>
                  <Link
                    href="/cart"
                  >
                    <a
                      aria-label="View your cart"
                      href="/cart"
                      onClick={[Function]}
                    >
                      View Your Bag
                      <span
                        style={
                          Object {
                            "marginLeft": "5px",
                            "marginRight": "5px",
                          }
                        }
                      >
                        (
                        0
                        )
                      </span>
                    </a>
                  </Link>
                  <Link
                    href=""
                  >
                    <a
                      aria-label="Go to checkout"
                      className="o-button o-button--negative o-button--lrg o-button--disabled"
                      href=""
                      onClick={[Function]}
                    >
                      CHECKOUT
                    </a>
                  </Link>
                </div>
              </MiniBag>
            </Connect(MiniBag)>
          </span>
        </div>
        <Connect(NavBar)>
          <NavBar
            categories={
              Object {
                "data": Array [],
                "error": false,
                "loading": false,
              }
            }
            dispatch={[Function]}
          >
            <div
              className="o-nav"
              role="navigation"
            />
          </NavBar>
        </Connect(NavBar)>
        <div
          className="o-body"
        >
          <CartTable
            cart={
              Object {
                "lineItems": Array [
                  Object {
                    "discount": 0,
                    "imageUrl": "",
                    "price": 10,
                    "productID": 1,
                    "productSku": "1231",
                    "quantity": 2,
                    "size": "size - 8",
                    "sku": "123",
                    "title": "test",
                  },
                ],
              }
            }
            updateQuantity={[Function]}
          >
            <section>
              <section
                className="c-cart-table__header"
              >
                <div
                  className="u-float--left"
                >
                  <h1>
                     My Bag 
                  </h1>
                </div>
                <div
                  className="u-float--right"
                >
                  <h4>
                     
                    <b>
                      1
                       
                      item
                    </b>
                     in your bag 
                  </h4>
                </div>
              </section>
              <section
                className="u-float-clear"
              >
                <LineItems
                  aria-label="Line Items"
                  cart={
                    Object {
                      "lineItems": Array [
                        Object {
                          "discount": 0,
                          "imageUrl": "",
                          "price": 10,
                          "productID": 1,
                          "productSku": "1231",
                          "quantity": 2,
                          "size": "size - 8",
                          "sku": "123",
                          "title": "test",
                        },
                      ],
                    }
                  }
                  updateQuantity={[Function]}
                >
                  <table
                    className="c-line-items"
                  >
                    <thead>
                      <tr
                        className="c-line-items__header"
                      >
                        <th
                          scope="col"
                        >
                          Your Items
                        </th>
                        <th
                          scope="col"
                        >
                          Size
                        </th>
                        <th
                          scope="col"
                        >
                          Quantity
                        </th>
                        <th
                          scope="col"
                        >
                          You save
                        </th>
                        <th
                          scope="col"
                        >
                          Subtotal
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr
                        className="c-line-items__column"
                        key="0"
                      >
                        <td
                          aria-label="test"
                        >
                          <Link
                            as="/products/1"
                            href="/products/product?1"
                          >
                            <a
                              href="/products/1"
                              onClick={[Function]}
                            >
                              <Image
                                alt="test"
                                aria-label="test"
                                className="c-line-items__image"
                                src=""
                              >
                                <div
                                  className="c-dummy_image c-line-items__image"
                                  style={
                                    Object {
                                      "height": undefined,
                                      "width": undefined,
                                    }
                                  }
                                />
                              </Image>
                              test
                            </a>
                          </Link>
                        </td>
                        <td>
                          size - 8
                        </td>
                        <td>
                          <select
                            data-variant="123"
                            onChange={[Function]}
                            value={2}
                          >
                            <option
                              aria-posinset={1}
                              aria-setsize={0}
                              key="0"
                              value={0}
                            >
                              0
                            </option>
                            <option
                              aria-posinset={2}
                              aria-setsize={1}
                              key="1"
                              value={1}
                            >
                              1
                            </option>
                            <option
                              aria-posinset={3}
                              aria-setsize={2}
                              key="2"
                              value={2}
                            >
                              2
                            </option>
                            <option
                              aria-posinset={4}
                              aria-setsize={3}
                              key="3"
                              value={3}
                            >
                              3
                            </option>
                            <option
                              aria-posinset={5}
                              aria-setsize={4}
                              key="4"
                              value={4}
                            >
                              4
                            </option>
                            <option
                              aria-posinset={6}
                              aria-setsize={5}
                              key="5"
                              value={5}
                            >
                              5
                            </option>
                            <option
                              aria-posinset={7}
                              aria-setsize={6}
                              key="6"
                              value={6}
                            >
                              6
                            </option>
                            <option
                              aria-posinset={8}
                              aria-setsize={7}
                              key="7"
                              value={7}
                            >
                              7
                            </option>
                            <option
                              aria-posinset={9}
                              aria-setsize={8}
                              key="8"
                              value={8}
                            >
                              8
                            </option>
                            <option
                              aria-posinset={10}
                              aria-setsize={9}
                              key="9"
                              value={9}
                            >
                              9
                            </option>
                            <option
                              aria-posinset={11}
                              aria-setsize={10}
                              key="10"
                              value={10}
                            >
                              10
                            </option>
                          </select>
                        </td>
                        <td>
                          -£
                          0
                        </td>
                        <td>
                          £
                          20
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </LineItems>
                <CartSummary
                  aria-label="Cart Summary"
                  cart={
                    Object {
                      "lineItems": Array [
                        Object {
                          "discount": 0,
                          "imageUrl": "",
                          "price": 10,
                          "productID": 1,
                          "productSku": "1231",
                          "quantity": 2,
                          "size": "size - 8",
                          "sku": "123",
                          "title": "test",
                        },
                      ],
                    }
                  }
                  updateQuantity={[Function]}
                >
                  <section
                    className="c-cart-summary"
                  >
                    <dl>
                      <dt>
                         Sub total: 
                      </dt>
                      <dd>
                         £
                        20
                         
                      </dd>
                    </dl>
                    <dl>
                      <dt>
                         Tax applied: 
                      </dt>
                      <dd
                        className="u-text-color--red"
                      >
                         +£
                        0
                         
                      </dd>
                    </dl>
                    <dl>
                      <dt>
                         Amount you saved on this purchase: 
                      </dt>
                      <dd
                        className="u-text-color--green"
                      >
                         -£
                        0
                         
                      </dd>
                    </dl>
                    <dl>
                      <dt>
                         Total amount to be paid: 
                      </dt>
                      <dd>
                         
                        <b>
                          £
                          20
                        </b>
                         
                      </dd>
                    </dl>
                  </section>
                </CartSummary>
              </section>
            </section>
          </CartTable>
        </div>
        <div
          className="o-footer"
        >
          Footer
        </div>
      </div>
    </Layout>
  </CartPage>
</Provider>
`;
